packages/react-next/src/Utilities/Customizer.tsx

```diff
@@ -0,0 +1,50 @@
+import * as React from 'react';
+import { useForceUpdate } from '@uifabric/react-hooks';
+import { ThemeProvider } from '@fluentui/react-theme-provider';
+import {
+  Customizations,
+  CustomizerContext,
+  mergeCustomizations,
+  ICustomizerContext,
+} from 'office-ui-fabric-react/lib/Utilities';
+import { convertLegacyTheme } from '../Styling/convertLegacyTheme';
+import { ICustomizerProps } from '../Utilities';
+
+/**
+ * Replacement for Customizer from utilities package. When theme is defined in settings,
+ * theme gets provided by ThemeProvider.
+ *
+ * TODO (xgao): deprecate settings prop.
+ */
+export const Customizer: React.FunctionComponent<ICustomizerProps> = props => {
+  const forceUpdate = useForceUpdate();
+
+  React.useEffect(() => {
+    Customizations.observe(forceUpdate);
+
+    return () => Customizations.unobserve(forceUpdate);
+  }, []);
+
+  const { contextTransform, children, disableThemeProvider } = props;
+  return (
+    <CustomizerContext.Consumer>
+      {(parentContext: ICustomizerContext) => {
```

Comment:
this there a way to avoid merging on every render?

Reply:
Maybe this can be memoized?

Reply:
memoization is the only way however it's not going to help.

I think most people are inlining the `settings` or `scopedSettings` props when they using the `Customizer`, as the following:
```
<Customizer settings={ theme: { fonts ... }} />
<Customizer scopedSettings={ Button: { styles: {root: ... } }} />
```

The good news is I don't think there is too big of a concern here. this "merge" logic is not expensive